core/capabilities/outgoing_connector_handler.go

```diff
@@ -89,8 +107,28 @@ func (c *ConnectorHandler) HandleGatewayMessage(ctx context.Context, gatewayID s
 	}
 	l.Debugw("handling gateway request")
 	switch body.Method {
-	case webapicapabilities.MethodWebAPITarget:
-		var payload webapicapabilities.TargetResponsePayload
+	case capabilities.MethodWebAPITarget:
+		var payload capabilities.TargetResponsePayload
+		err := json.Unmarshal(body.Payload, &payload)
+		if err != nil {
+			l.Errorw("failed to unmarshal payload", "err", err)
+			return
+		}
+		c.responseChsMu.Lock()
+		defer c.responseChsMu.Unlock()
+		ch, ok := c.responseChs[body.MessageId]
+		if !ok {
+			l.Errorw("no response channel found")
+			return
+		}
+		select {
+		case ch <- msg:
+			delete(c.responseChs, body.MessageId)
+		case <-ctx.Done():
+			return
+		}
+	case capabilities.MethodComputeAction:
```

Comment:
I know you talked about working on more deduplication, but I wonder if these two cases can be combined somehow?

Reply:
exactly, this is part of what I mentioned, basically these methods are the same, they would receive the payload type as a parameter 👍🏼